### PR TITLE
Add Detox

### PR DESCRIPTION
**Pull request by asafkorem**
# 📝 Why & how

# ✅ Checklist

- [x] Added library to **`react-native-libraries.json`**
- [ ] Updated library in **`react-native-libraries.json`**

- [ ] Documented in this PR how to use the feature or replicate the bug.
- [ ] Documented in this PR how you fixed or created the feature.

### CODE REVIEW

File: react-native-libraries.json [hunk 1]
```diff
@@ -8590,5 +8590,18 @@
...
+    "npmPkg": "detox",
+    "nameOverride": "Detox",
+    "examples": [
+      "https://github.com/wix/Detox/tree/master/examples"
+    ],
+    "images": [
+      "https://github.com/wix/Detox/raw/master/docs/img/Detox.gif"
+    ],
+    "ios": true,
+    "android": true
```
Comment (Simek) [inline react-native-libraries.json:8605]:
```suggestion
    "android": true,
    "dev": true
```

Let's also add `dev` flag.

File: react-native-libraries.json [hunk 2]
```diff
@@ -8590,5 +8590,18 @@
     "ios": true,
     "android": true,
     "newArchitecture": true
+  },
+  {
+    "githubUrl": "https://github.com/wix/Detox",
+    "npmPkg": "detox",
+    "nameOverride": "Detox",
+    "examples": [
+      "https://github.com/wix/Detox/tree/master/examples"
```
Comment (Simek) [inline react-native-libraries.json:8599]:
```suggestion
      "https://github.com/wix/Detox/tree/master/examples/demo-react-native",
      "https://github.com/wix/Detox/tree/master/examples/demo-react-native-detox-instruments",
      "https://github.com/wix/Detox/tree/master/examples/demo-react-native-jest"
```

Let's list individual example apps instead.

**Review by Simek — CHANGES_REQUESTED** [review]
Thank you for adding new entry to the directory! 👍 

Left two small suggestion, besides that LGTM.

File: react-native-libraries.json (hunk 2 above)
Comment (asafkorem) [reply react-native-libraries.json]: To be honest, we have plans to delete some of the projects in the list, and replace them with new ones.
I replaced the folder name with a specific app, and will update this list once when we'll have the new projects.

**Comment by asafkorem** [comment]
Thanks, done :+1:

**Review by Simek — APPROVED** [review]
Thank you for the updates! 🙏

**Pull request merged**